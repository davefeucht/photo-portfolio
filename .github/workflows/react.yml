# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
