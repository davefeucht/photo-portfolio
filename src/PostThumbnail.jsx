import React from "react";
import axios from "axios";
import PropTypes from "prop-types";

export default class PostThumbnail extends React.Component {
  constructor(props) {
    super(props);

    this.propTypes = {
      id: PropTypes.number,
      image: PropTypes.number,
      site: PropTypes.string,
      clickImage: PropTypes.func
    };

    this.state = {
      thumbnailUrl: ""
    };
  }

  _getPostThumbnail() {
    axios.get("http://" + this.props.site + "/wp-json/wp/v2/media/" + this.props.image + "/")
      .then(res => {
        let thumbnailImage = new Image();
        const thumbnailUrl = res.data.media_details.sizes.thumbnail.source_url;
        this.setState({ thumbnailUrl });
        thumbnailImage.src = thumbnailUrl;
      }, error => {
        const errorMsg = "Did not work: " + (error.response ? error.response : error);
        this.setState({ errorMsg });
      });
  } 

  _showFullPost(event) {
    event.preventDefault();
    this.props.clickImage(this.props.id);
  }

  componentWillMount() {
    this._getPostThumbnail();
  }

  render() {
    let divStyle = {backgroundImage: "url(" + this.state.thumbnailUrl + ")", backgroundRepeat: "no-repeat", backgroundPosition: "center center", backgroundSize: "auto 100%"};

    return(
      <div className="post-thumbnail" style={divStyle} onClick={this._showFullPost.bind(this)}>
      </div>
    );
  }
}