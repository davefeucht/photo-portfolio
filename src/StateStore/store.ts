import { configure, makeAutoObservable } from 'mobx';
import { ImageData, Image, Store, Category, Post, Page } from '../utils/types';

configure({ enforceActions: 'observed' });

class stateStore implements Store {
    menuState = 'closed'
    applicationRoot: HTMLElement = null;
    screenInfo: {
        width: number,
        height: number
    }
    siteInfo: {
        siteName: string,
        siteUrl: string
    }
    visiblePost: {
        postId: number,
        postTitle: string,
        tags: number[],
        tagNames: string[],
        fullImageUrl: string,
        width: number,
        height: number
    };
    categoryList: Category[] = [];
    currentCategoryPosts: Post[] = [];
    currentCategoryData: Category = {
        id: 0,
        count: 0,
        description: '',
        link: '',
        name: '',
        slug: '',
        taxonomy: '',
        parent: 0
    };
    pages: Page[] = [];
    currentPageData: Page = {
        id: 0,
        link: '',
        modified: '',
        slug: '',
        type: '',
        title: {
            rendered: ''
        },
        content: {
            rendered: '',
            protected: false
        },
        author: 0,
        featured_media: 0
    };
    maxItemsPerPage = 10;

    constructor() {
        makeAutoObservable(this);
    }

    setMenuState = (state: string) => {
        this.menuState = state;
    }

    toggleMenuState = () => {
        this.menuState = this.menuState === 'closed' ? 'open' : 'closed';
    }

    setApplicationRoot = (element: HTMLElement) => {
        this.applicationRoot = element;
    }

    setSiteName = (name: string) => {
        this.siteInfo.siteName = name;
    }

    setVisiblePost = (postId: number, postTitle: string) => {
        this.visiblePost.postId = postId;
        this.visiblePost.postTitle = postTitle;
    }

    setVisiblePostImage = (fullImageUrl: string) => {
        this.visiblePost.fullImageUrl = fullImageUrl;
    }

    setVisiblePostTags = (tagNames: string[]) => {
        tagNames.forEach(tagName => {
            this.visiblePost.tagNames.push(tagName);
        });
    }

    clearVisiblePostTagNames = () => {
        this.visiblePost.tagNames.splice(0, this.visiblePost.tagNames.length);
    }

    setCategoryList = (categories: Category[]) => {
        this.categoryList.length = 0;
        categories.forEach((category, index) => {
            this.categoryList[index] = category;
        })
    }

    setCategoryPosts = (posts: Post[]) => {
        this.currentCategoryPosts.length = 0;
        posts.forEach((post, index) => {
            this.currentCategoryPosts[index] = post;
        })
    }

    setCategoryData = (categoryData: Category) => {
        Object.keys(categoryData).forEach((key: keyof Category) => {
            this.currentCategoryData[key] = categoryData[key];
        });
    }

    setThumbnailImageUrl = (imageData: ImageData) => {
        this.currentCategoryPosts[imageData.post_index].thumbnail_image = imageData.image_url;
    }

    setCurrentPost = (postData: Post) => {
        Object.keys(postData).forEach((key: keyof Image) => {
            this.visiblePost[key] = postData[key];
        })
    }

    setPages = (pages: Page[]) => {
        pages.forEach(page => {
            this.pages.push(page);
        });
    }

    setPageData = (pageData: Page) => {
        Object.assign(this.currentPageData, pageData);
    }
}

export default stateStore;
