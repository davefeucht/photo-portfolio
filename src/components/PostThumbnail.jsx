import React from "react";
import axios from "axios";
import PropTypes from "prop-types";
import StyledPostThumbnail from "./styledComponents/StyledPostThumbnail.jsx";

export default class PostThumbnail extends React.Component {
  static propTypes = {
    id: PropTypes.number,
    image: PropTypes.number,
    site: PropTypes.string,
    clickImage: PropTypes.func
  };

  state = {
    thumbnailUrl: ""
  };

  _getPostThumbnail() {
    axios.get("http://" + this.props.site + "/wp-json/wp/v2/media/" + this.props.image + "/")
      .then(res => {
        let thumbnailImage = new Image();
        const thumbnailUrl = res.data.media_details.sizes.medium.source_url;
        this.setState({ thumbnailUrl });
        thumbnailImage.src = thumbnailUrl;
      }, error => {
        const errorMsg = "Did not work: " + (error.response ? error.response : error);
        this.setState({ errorMsg });
      });
  } 

  _showFullPost(event) {
    event.preventDefault();
    this.props.clickImage(this.props.id);
  }

  componentWillMount() {
    this._getPostThumbnail();
  }

  render() {
    const divStyle = {backgroundImage: "url(" + this.state.thumbnailUrl + ")", backgroundRepeat: "no-repeat", backgroundPosition: "center center", backgroundSize: "cover"};

    return(
      <StyledPostThumbnail style={divStyle} onClick={this._showFullPost.bind(this)} />
    );
  }
}