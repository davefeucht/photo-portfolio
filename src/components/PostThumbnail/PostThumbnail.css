div {
  height: 100%;
  cursor: pointer;
}