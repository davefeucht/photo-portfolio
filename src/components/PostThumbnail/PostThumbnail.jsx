/****************
* PostThumbnail component displays the thumbnail image for a Post
****************/

import React from "react";
import { runInAction } from "mobx";
import { observer } from 'mobx-react';
import './PostThumbnail.css';

const PostThumbnail = observer(({ stateStore, id, title, index }) => {
  const _showFullPost = () => {
    runInAction(() => {
      stateStore.setVisiblePost(id, title);
      stateStore.setShowAllPosts(false);
    });
  }

  const divStyle = {backgroundImage: "url(" + (stateStore.currentCategoryPosts[index].thumbnail_image ? stateStore.currentCategoryPosts[index].thumbnail_image : "") + ")", backgroundRepeat: "no-repeat", backgroundPosition: "center center", backgroundSize: "cover"};

  return(
    <div className="post-thumbnail" style={divStyle} onClick={_showFullPost.bind(this)}></div>
  );
});

PostThumbnail.displayName = 'PostThumbnail';

export default PostThumbnail;