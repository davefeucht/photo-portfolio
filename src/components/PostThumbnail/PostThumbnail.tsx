/** **************
* PostThumbnail component displays the thumbnail image for a Post
*************** */

import './PostThumbnail.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { Link, useLocation } from 'react-router-dom';

interface PostThumbnailProps {
    id: string,
    thumbnailImage: string
}

/*
 * currentCategoryPosts
*/

const PostThumbnail: React.FC<PostThumbnailProps> = ({ id, thumbnailImage = '' }) => {
    const divStyle = { backgroundImage: `url(${thumbnailImage})` };
    const location = useLocation();

    return (
        <Link
            to={{ pathname: `post/${id}` }}
            state={{ background: location }}
        >
            <div className="post-thumbnail" style={divStyle} />
        </Link>
    );
};

PostThumbnail.displayName = 'PostThumbnail';

export default observer(PostThumbnail);
