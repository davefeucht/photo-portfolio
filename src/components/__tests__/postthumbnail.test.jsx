import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import StateStore from '../../StateStore/store';
import PostThumbnail from '../PostThumbnail/PostThumbnail';

const categoryPosts = [
    {
        thumbnail_image: 'some_url'
    }
];

const postId = 35;

test('PostThumbnail displays', () => {
    const store = new StateStore();
    store.setCategoryPosts(categoryPosts);
    const { container } = render(
        <MemoryRouter>
            <PostThumbnail stateStore={store} id={postId} index={0} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
});
