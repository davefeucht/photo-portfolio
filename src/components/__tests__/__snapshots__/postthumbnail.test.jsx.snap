// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostThumbnail displays 1`] = `
<a
  href="/post/35"
>
  <div
    class="post-thumbnail"
    style="background-image: url(some_url);"
  />
</a>
`;
